.buildkite/steps/tests.sh

```diff
@@ -8,14 +8,15 @@ go install gotest.tools/gotestsum@v1.8.0
 
 go install github.com/buildkite/test-engine-client@v1.5.0-rc.3
 
-echo '+++ Running tests'
 export BUILDKITE_TEST_ENGINE_SUITE_SLUG=buildkite-agent
 export BUILDKITE_TEST_ENGINE_TEST_RUNNER=gotest
 export BUILDKITE_TEST_ENGINE_RESULT_PATH="junit-${BUILDKITE_JOB_ID}.xml"
 export BUILDKITE_TEST_ENGINE_RETRY_COUNT=1
-export BUILDKITE_TEST_ENGINE_TEST_CMD="gotestsum --junitfile={{resultPath}} -- -count=1 -coverprofile=cover.out $@ {{packages}}"
+if [[ "${TEST_COVERAGE:-false}" == "true" ]]; then
+  mkdir -p coverage
+  export BUILDKITE_TEST_ENGINE_TEST_CMD="gotestsum --junitfile={{resultPath}} -- -count=1 -cover $* {{packages}} -test.gocoverdir=$PWD/coverage"
+else
+  export BUILDKITE_TEST_ENGINE_TEST_CMD="gotestsum --junitfile={{resultPath}} -- -count=1 $* {{packages}}"
+fi
```

Comment:
Can we continue having coverage for Windows too?

I think it's a matter of translating Unix paths back into Windows paths...

Does this work?

```suggestion
mkdir -p coverage
COVERAGE_DIR="$PWD/coverage"
if [[ "$(uname)" == *"windows"* ]]; then
  COVERAGE_DIR="$(cygpath -w "${COVERAGE_DIR}")"
fi
export BUILDKITE_TEST_ENGINE_TEST_CMD="gotestsum --junitfile={{resultPath}} -- -count=1 -cover $* {{packages}} -test.gocoverdir=${COVERAGE_DIR}"
```

Reply:
I tried a number of ways to get it works on Windows, none of them works including your suggestion 😬 .

They all failed on: https://buildkite.com/buildkite/agent/builds/10110/steps?jid=01968a3e-200d-4ff3-a51e-7d5098c9a90c

So I gave up and make this opt-in.

Reply:
I suggested it without trying it, but I think the problem may be that `uname` doesn't actually return a string containing "windows" on Windows:

<img width="270" alt="Screenshot 2025-05-01 at 3 29 41 pm" src="https://github.com/user-attachments/assets/97c3b9fc-c692-4fdd-b597-66f0c02fe2aa" />

Doing some [reading](https://stackoverflow.com/questions/3466166/how-to-check-if-running-in-cygwin-mac-or-linux) it could be `CYGWIN*`, `MINGW*`, or `MSYS*`?

Reply:
Another complication of including windows and other platform tests is that: there seems to be some form of interference among them: 

`error: counter mode clash while reading meta-data file coverage/covmeta.15abce4185690bdced71054bf6cf38cd: previous file had atomic, new file has set`

Reply:
I wonder why it's failing that way, when the Windows tests in that build all failed to write coverage profiles? ☹️

Reply:
I am not sure either. 
I don't know what the "previous file" was referring to. I have a guess, it was referring to other files about the same test records? and somehow, some are using "set" mode, others are using "atomic" mode.

But I don't know if I should drill too deep. I wonder if it would be possible to kick this can down the road?

Reply:
What if we generated separate coverage reports for each platform? That should correspond to how things used to be, and hopefully avoid this apparent counter mode clash error?

At this point I don't mind if you do it with or without Windows.

Reply:
can do, but the amd64, arm64, and race are going to have basically the same data?

Reply:
cool, I created one CI step each for every test step (except for Windows). 

I gave Windows another stab (I fixed the uname check) and still no luck, the feedback cycle is bit too long for me, I don't have a windows env to test this properly. 😢